libsolidity/interface/CompilerStack.cpp

```diff
@@ -1352,14 +1353,30 @@ void CompilerStack::generateEVMFromIR(ContractDefinition const& _contract)
 	//cout << yul::AsmPrinter{}(*stack.parserResult()->code) << endl;
 
 	// TODO: support passing metadata
-	// TODO: use stack.assemble here!
-	yul::MachineAssemblyObject init;
-	yul::MachineAssemblyObject runtime;
-	std::tie(init, runtime) = stack.assembleWithDeployed(IRNames::deployedObject(_contract));
-	compiledContract.object = std::move(*init.bytecode);
-	compiledContract.runtimeObject = std::move(*runtime.bytecode);
-	// TODO: refactor assemblyItems, runtimeAssemblyItems, generatedSources,
-	//       assemblyString, assemblyJSON, and functionEntryPoints to work with this code path
+
+	tie(compiledContract.evmAssembly, compiledContract.evmRuntimeAssembly) = stack.assembleEVM(IRNames::deployedObject(_contract));
+	solAssert(compiledContract.evmAssembly, "");
+	try
+	{
+		// Assemble deployment (incl. runtime) object.
+		compiledContract.object = compiledContract.evmAssembly->assemble();
+	}
+	catch(evmasm::AssemblyException const&)
```

Comment:
```suggestion
	catch (evmasm::AssemblyException const&)
```

Reply:
So this is actually a copy of `compileContract` with the same style. I'll need to think about a way to deduplicate the code.